src/screentovnc.cpp

```diff
@@ -515,20 +515,82 @@ void ScreenToVnc::rfbProcessTrigger()
 
 void ScreenToVnc::init_fingerPointers()
 {
-    pointerFingerPtr = rfbMakeXCursor(pointer_finger.width,
-                                      pointer_finger.height,
-                                      pointer_finger.bitmask,
-                                      pointer_finger.bitmask);
-
-    pointerFingerTouchPtr = rfbMakeXCursor(pointer_finger_touch.width,
-                                           pointer_finger_touch.height,
-                                           pointer_finger_touch.bitmask,
-                                           pointer_finger_touch.bitmask);
-
-    emptyMousePtr = rfbMakeXCursor(empty_mouse.width,
-                                   empty_mouse.height,
-                                   empty_mouse.bitmask,
-                                   empty_mouse.bitmask);
+    emptyMouseCursorInfo = ScreenToVnc::load_cursor_info_from_png(
+        "/usr/share/lipstick2vnc/cursor_empty.png");
+    pointerCursorInfo = ScreenToVnc::load_cursor_info_from_png(
+        "/usr/share/lipstick2vnc/cursor_pointer.png");
+    pointerTouchCursorInfo = ScreenToVnc::load_cursor_info_from_png(
+        "/usr/share/lipstick2vnc/cursor_pointer_touch.png");
+
+    pointerFingerPtr = rfbMakeXCursor(pointerCursorInfo.width,
+        pointerCursorInfo.height,
+        pointerCursorInfo.bitmask,
+        pointerCursorInfo.bitmask);
+
+    pointerFingerTouchPtr = rfbMakeXCursor(pointerTouchCursorInfo.width,
+        pointerTouchCursorInfo.height,
+        pointerTouchCursorInfo.bitmask,
+        pointerTouchCursorInfo.bitmask);
+
+    emptyMousePtr = rfbMakeXCursor(emptyMouseCursorInfo.width,
+        emptyMouseCursorInfo.height,
+        emptyMouseCursorInfo.bitmask,
+        emptyMouseCursorInfo.bitmask);
+}
+
+
+cursor_info ScreenToVnc::load_cursor_info_from_png(const char* filename)
+{
+    cursor_info info;
+    std::vector<unsigned char> png;
+    unsigned long width;
+    unsigned long height;
+
+    int bpp = 4;
+
+    QImage pngImg = QImage(filename);
+    width = pngImg.width();
+    height = pngImg.height();
+
+    png.reserve(bpp * width * height);
+
+    for(int y=0; y<height; y++){
+      QRgb *pxRow = (QRgb*) pngImg.scanLine(y);
+      for(int x=0; x<width; x++) {
+          QRgb &px = pxRow[x];
+          png.push_back(qRed(px));
+          png.push_back(qGreen(px));
+          png.push_back(qBlue(px));
+          png.push_back(qAlpha(px));
+      }
+    }
+
+    info.width = width;
+    info.height = height;
+    info.pixel_data = png;
+    info.bitmask = (char*) malloc(sizeof(char) * width * height + 1);
```

Comment:
Never freed. Maybe having a destructor on cursor_info could be a simple way? I.e. cursor_info having ownership here.

Reply:
simply putting the free in a ~cursor_info() breaks it, it gets free'd too early and the png is drawn with a random bitmask.
since the malloc in init_fingerPointers() is called from the ScreenToVnc() ctor, i just put the free in that already-present destructor.